framework/contracts/native/ica-client/src/queries.rs

```diff
@@ -0,0 +1,385 @@
+use crate::state::{Config, CONFIG};
+use abstract_ica::{msg::ConfigResponse, ChainType, IcaAction, IcaActionResponse};
+use abstract_std::objects::TruncatedChainId;
+use cosmwasm_std::{ensure_eq, CosmosMsg, Deps, Env};
+
+use crate::{chain_types::evm, contract::IcaClientResult, error::IcaClientError};
+
+/// Timeout in seconds
+pub const PACKET_LIFETIME: u64 = 60 * 60;
+
+pub fn config(deps: Deps) -> IcaClientResult<ConfigResponse> {
+    let Config {
+        version_control,
+        ans_host,
+    } = CONFIG.load(deps.storage)?;
+    Ok(ConfigResponse {
+        ans_host: ans_host.address.into_string(),
+        version_control_address: version_control.address.into_string(),
+    })
+}
+
+pub(crate) fn ica_action(
+    deps: Deps,
+    env: Env,
+    _proxy_address: String,
+    chain: TruncatedChainId,
+    mut actions: Vec<IcaAction>,
+) -> IcaClientResult<IcaActionResponse> {
+    // match chain-id with cosmos or EVM
+    use abstract_ica::CastChainType;
+    let chain_type = chain.chain_type().ok_or(IcaClientError::NoChainType {
+        chain: chain.to_string(),
+    })?;
+
+    // todo: what do we do for msgs that contain both cosmos and EVM messages?
+    // Best to err if there's conflict.
+
+    // sort actions
+    // 1) Transfers
+    // 2) Calls
+    // 3) Queries
+    actions.sort_unstable();
+
+    let cfg = CONFIG.load(deps.storage)?;
+
+    // TODO: should we match ChainType first, so it doesn't get so nested?
```

Comment:
Yes because then each ChainType's functionality can be separated into functions more clearly as well.

Reply:
The type of chain is inside of the action so then assertions need to be moved. Let's refactor it once we support another ICA impl, then it'll be clearer how to structure it.